modules/repository/repo.go

```diff
@@ -127,11 +127,24 @@ func MigrateRepositoryGitData(ctx context.Context, u *models.User, repo *models.
 	}
 
 	if opts.Mirror {
+		var interval = setting.Mirror.DefaultInterval
+		if opts.MirrorInterval != "" {
+			interval, err = time.ParseDuration(opts.MirrorInterval)
+			if err != nil {
+				log.Error("Failed to set Interval: %v", err)
+				return repo, err
+			}
+			if interval != 0 && interval < setting.Mirror.MinInterval {
```

Comment:
can parseduration result in negative time?

Reply:
Yes it can, however with the current code it returns the error: "Interval is set below Minimum Interval" to the client. In the logs this is produced:
```
2020/12/30 22:16:28 ...s/repository/repo.go:139:MigrateRepositoryGitData() [E] Interval: -24h is too frequent
2020/12/30 22:16:28 ...i/v1/repo/migrate.go:229:handleMigrateError() [E] MigrateRepository: Interval is set below Minimum Interval
```

Reply:
D'oh!